### PR TITLE
feat: chat legend enhancement #2277

### PR DESCRIPTION
**Pull request by YadhunandanGowda**
Issue: #2277 

- I have implemented the Alignment functionality.
- After the approval of this PR I will raise one more PR which contains 'Display' functionality.

[screen-capture (2).webm](https://github.com/awslabs/iot-app-kit/assets/65999096/e192bf30-86ad-4ef6-a5e9-d2b14750db5d)

### CODE REVIEW

**Comment by diehbria** [comment]
is there a required change in `iot-application`'s verification of dashboard definition to allow this change to be merged in without breaking users experience of IoT Dashboard application?

File: packages/react-components/src/hooks/useECharts/useResizeableEChart.ts [hunk 1]
```diff
@@ -34,44 +40,54 @@ const getChartWidth = (width: number, staticWidth: number, rightLegend?: ChartLe
...
   const { width, height } = size;
   const [leftLegendRef, { width: leftLegendWidth }] = useMeasure<HTMLDivElement>();
   const [chartWidth, setChartWidth] = useState(getChartWidth(width, leftLegendWidth));
+  const [chartHeight, setChartHeight] = useState(getChartHeight(height));
   const rightLegendWidth = rightLegend ? width - leftLegendWidth - chartWidth : 0;
+  const rightLegendHeight = rightLegend ? height - chartHeight : 0;
 
   const onResize = (_event: SyntheticEvent, data: ResizeCallbackData) => {
     _event.stopPropagation();
+    console.log('***', data.size);
```
Comment (mnischay) [inline packages/react-components/src/hooks/useECharts/useResizeableEChart.ts:55]: seems like a leftover ?

**Review by mnischay — COMMENTED** [review]
the flow seems to be good, just some minot nits

File: packages/dashboard/src/customization/propertiesSections/components/alignmentDropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+const defaultOption = {
+  label: 'Right',
+  value: 'right',
+};
+
+const { legendAlignmentOptions, defaultValue } = dropdownConsts.legendAlignment;
+
+export const AlignmentDropdown: FC<AlignmentDropdownProps> = ({ disabled = false, position, onTypeChange }) => {
+  const handleType = (event: NonCancelableCustomEvent<SelectProps.ChangeDetail>) => {
+    onTypeChange(event.detail.selectedOption.value || defaultValue.value);
```
[the remark this answers is not in the document]
Comment (YadhunandanGowda) [reply packages/dashboard/src/customization/propertiesSections/components/alignmentDropdown.tsx]: took the reference from 'packages/dashboard/src/customization/propertiesSections/components/lineStyleDropdown.tsx' and rectified it.

File: packages/dashboard/src/customization/propertiesSections/components/alignmentDropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  };
+
+  return (
+    <FormField label='Alignment'>
+      <Select
+        disabled={disabled}
+        selectedOption={Object.values(legendAlignmentOptions).find(({ value }) => value === position) || defaultOption}
```
[the remark this answers is not in the document]
Comment (YadhunandanGowda) [reply packages/dashboard/src/customization/propertiesSections/components/alignmentDropdown.tsx]: the find is to get the selected value as an object from the list. we have to pass the object not just a value. took the reference from 'packages/dashboard/src/customization/propertiesSections/components/lineStyleDropdown.tsx'

File: packages/dashboard/src/customization/propertiesSections/constants.ts [hunk 4]
```diff
@@ -0,0 +1,86 @@
+import type { SelectProps } from '@cloudscape-design/components';
```
[the remark this answers is not in the document]
Comment (YadhunandanGowda) [reply packages/dashboard/src/customization/propertiesSections/constants.ts:1]: Thank you.

**Comment by YadhunandanGowda** [comment]
Hi @mnischay 
As I checked with the team, we don't have to make any necessary changes in 'iot-application'.
Thank you.

**Comment by tracy-french** [comment]
@mnischay Is OOTO this week. Assuming this change is good based on previous feedback, updates, and responses.

**Review by tracy-french — APPROVED** [review]

**Pull request merged**